### ISSUE TITLE
Effects.receive / Effects.send

### ISSUE THREAD

**Issue by Gozala** [evancz/elm-effects#31]
Creating an effect that just triggers an action currently is verbose and requires users to also understand tasks. I would like to propose a syntax sugar mostly to defer requirement of understanding of tasks until later point in the learning curve.

How about adding such a `receive` or `send` function to Effects module:

```elm
receive : a -> Effects a
receive = action -> Effects.task(Task.succeed(action))
```
*Relevant thread: https://groups.google.com/forum/?fromgroups#!topic/elm-discuss/ys0hds5rsWs*

**Comment by jvoigtlaender** [comment]
I'm not sure I buy the motivation of deferring to learn about tasks.

Can you give a concrete example where you would want to use `Effects.receive` and where the programmer does not already need to know about tasks anyway?

Note that [this comment](https://github.com/evancz/elm-effects/issues/27#issuecomment-163135893) might obviate some potential uses of `Effects.receive`. I actually conjecture it obviates all uses of `Effects.receive` in situations where the programmer does not yet use tasks anyway.

**Comment by Gozala** [comment]
@jvoigtlaender In an application I'm working with a small group of people we do a lot of animations in return to user interaction driven by `Effects.tick`, at the end of the animation we do usually trigger an action associated with a user interaction that triggered animation in first place. There are also handful of cases where actions triggered at the end of the animation are handled higher up in the component tree, meaning that it's not actually component that has send action does state update / io but an owner n levels above it.

Such a use I am afraid is not handled by a suggestion you made in #27

**Comment by Gozala** [comment]
It's little late here and I'm having trouble staying on point, I apologize.

What I was trying to say in previous comment is that it was easy enough for me to explain general idea of sending an action caused by user interaction at the end of animation so that desired update will happen at the end of animation. Part that made it really difficult was when we had to use `Task`s to do so which required explaining tasks, not only that people who though were understanding effects got confused about them as well. Also note that they had no problem understanding `Effects.tick` as it simply schedules action to be send back in on next animation frame.

**Comment by jvoigtlaender** [comment]
Thanks, that makes perfect sense. I had been wondering in what situations one would be in `Effects` land, and want to do something like `Effects.receive`, but have not also encountered `Effects.task` along the way already. I had forgotten about `Effects.tick` here. But yes, that's a situation where one would already deal with `Effects` to a non-trivial degree, and still not have had need to deal with tasks.

So yes, I'm on board with:
* If one sometimes needs the functionality of `Effects.task << Task.succeed`, then it makes sense to introduce a helper function for that which does not mentally involve tasks.
* The mental model would be that `Effects.receive action` is like `Effects.tick (always action)`, except that there is no waiting until the next animation frame. (I think you mentioned this already on the mailing list or so. Anyway: Maybe that relationship to `Effects.tick` can inform a better choice of name than `Effects.receive`?)

How often one actually needs the functionality of `Effects.task << Task.succeed` is another issue I guess, the one discussed in the other thread.

**Comment by jvoigtlaender** [comment]
Some possible names:
* `Effects.send`
* `Effects.succeed`
* `Effects.pure`
* `Effects.return`
* `Effects.action`
* `Effects.next`
* `Effects.issue`
* `Effects.pass`
* `Effects.push`
* `Effects.put`
* `Effects.post`
* `Effects.emit`

**Comment by evancz** [comment]
**About process:** A motivating example needs to be in the initial comment. It's fine that it exists somewhere else on the internet, but just put it in this thread please. If you see an issue that is not this way, make it this way.

There are a number of viable proposals to elm-effects right now. The `silentTask` idea #28 legitimately needs support here, it cannot be done as a helper function. This proposal can easily be done as a helper though.

I have been talking with @Gozala offline separately, and I expect there will be at least one more "fundamental" thing in elm-effects for when you want a task to run synchronously (i.e. without going through the event loop before starting). I would like to make all of these API changes at the same time so we can get a coherent set of names and analogies across the whole API.

So my advice would be, write a helper function locally that your team can use. This will be in consideration for the next batch of changes and the  name will be chosen such that things feel coherent with the other more serious changes I have in mind. This will be part of [this](https://groups.google.com/forum/#!topic/elm-dev/1JW6wknkDIo) and [this](https://groups.google.com/forum/#!topic/elm-dev/3IpHt1nMHqA), so it is very high on my list.
